### PR TITLE
Adjustments for end to end tests

### PR DESCRIPTION
**Pull request by endzyme**
- Refactored all end to end tests into a file
- Added specific end to end course files to test specific behavior in isolation
- Added structure for future tests and output to show which tests fail
- Feature Added: ALL tests will run every time, not just fail on the first broken test
- Added light "framework" for adding tests and reporting errors

### CODE REVIEW

**Comment by endzyme** [comment]
Downside to this change are:
- The end to end tests now take an additional minute or so to run
- Little more complexity in writing end to end tests
- More yaml files for course tests

Upsides:
- Better isolation in end to end testing of features
- An easier way to read/write tests
- Allows us to test exit codes now

File: end_to_end_testing/setup.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+echo "Installing git and jq"
+sudo apt-get install -yqq jq git
+
+echo "Installing Reckoner"
+sudo cp /tmp/binaries/reckoner-linux-amd64 /usr/local/bin/reckoner
+reckoner version
+
+echo "Installing Kind"
+curl -sLO https://github.com/kubernetes-sigs/kind/releases/download/0.2.1/kind-linux-amd64
```
Comment (sudermanjr) [inline end_to_end_testing/setup.sh]: Should we update this to 0.4.0 now?  Also, should we pin the Kubernetes version we are using for this?  I think 0.4.0 now defaults to 1.14 or something like that.

Comment (endzyme) [reply end_to_end_testing/setup.sh]: I will adjust that in a new PR if that's ok - good call out and I will make sure to Pin both the Kind version to 0.4.0 and K8s version to v1.14.x

**Comment by bambash** [comment]
+1 on pinning the kubernetes and kind versions

:+1: looks great!

**Pull request merged**